### PR TITLE
Adds precision flag for f32 or f16 precision

### PR DESCRIPTION
**Pull request by IanNod**
- Adds precision flag for stateless_llama and gen_external_params
 - Saves f16 or f32 weights to external params from gen_external_params
 - Saves IR/vmfb for f16 or f32 model from stateless_llama

### CODE REVIEW

File: python/turbine_models/custom_models/stateless_llama.py [hunk 1]
```diff
@@ -227,6 +234,8 @@ def forward(token0: torch.Tensor, *state0_flat):
         )
         with open(f"{safe_name}.vmfb", "wb+") as f:
             f.write(flatbuffer_blob)
+        print("saved to ", safe_name + ".vmfb")
+        exit()
```
Comment (stellaraccident) [inline python/turbine_models/custom_models/stateless_llama.py:238]: Extra debugging to remove?

File: python/turbine_models/gen_external_params/gen_external_params.py [hunk 2]
```diff
@@ -12,14 +12,17 @@
     default="meta-llama/Llama-2-7b-chat-hf",
 )
 parser.add_argument("--quantization", type=str, default="int4")
-parser.add_argument("--weight_path", type=str, default="Llama2_7b_i4quant.safetensors")
+parser.add_argument("--weight_path", type=str, default="Llama2_7b.safetensors")
```
Comment (stellaraccident) [inline python/turbine_models/gen_external_params/gen_external_params.py:15]: It's going to become increasingly hard to keep these straight. I would suggest deriving the default from the other parameters and including the precision in the default filename. Probably want to have it default to None and then handle that at the usage site.

**Review by stellaraccident — APPROVED** [review]
A couple of small nits either for now or later.

File: python/turbine_models/custom_models/stateless_llama.py (hunk 1 above)
Comment (IanNod) [reply python/turbine_models/custom_models/stateless_llama.py:238]: This is more of a quick and dirty fix to an error not returning anything back to the main function when compiling to vmfb. We are done at this point so I'm just exiting, figured true cleanup would wait for the refactor

**Pull request merged**